test/manual/s3_test.cc

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+
+#include <boost/test/unit_test.hpp>
+#include <seastar/core/thread.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/file.hh>
+#include <seastar/core/fstream.hh>
+#include <seastar/testing/test_case.hh>
+#include "utils/s3/client.hh"
+
+/*
+ * Tests below expect minio server to be running on localhost
+ * with the bucket named "testbucket" created with unrestricted
+ * anonymous read-write access
+ */
+
+static const ipv4_addr s3_server("127.0.0.1", 9000);
+
+SEASTAR_TEST_CASE(test_client_put_get_object) {
+    return seastar::async([] {
+        const sstring name("/testbucket/testobject");
+
+        fmt::print("Make client\n");
+        auto cln = s3::client::make(s3_server);
+
+        fmt::print("Put object {}\n", name);
+        temporary_buffer<char> data = sstring("1234567890").release();
+        cln->put_object(name, std::move(data)).get();
+
+        fmt::print("Get object size\n");
+        size_t sz = cln->get_object_size(name).get0();
+        BOOST_REQUIRE_EQUAL(sz, 10);
+
+        fmt::print("Get object content\n");
+        temporary_buffer<char> res = cln->get_object_contiguous(name).get0();
+        BOOST_REQUIRE_EQUAL(to_sstring(std::move(res)), sstring("1234567890"));
+
+        fmt::print("Get object part\n");
+        res = cln->get_object_contiguous(name, s3::range{ 1, 3 }).get0();
+        BOOST_REQUIRE_EQUAL(to_sstring(std::move(res)), sstring("234"));
+
+        fmt::print("Delete object\n");
+        cln->delete_object(name).get();
+
+        fmt::print("Verify it's gone\n");
+        try {
+            cln->get_object_size(name).get();
+            BOOST_REQUIRE(false);
+        } catch (const std::exception& e) {
+            fmt::print(" -> {}\n", e.what());
+        }
```

Comment:
alternatively, we can do something like:

BOOST_REQUIRE_EXCEPTION(cln->get_object_size(name).get(), std::runtime_error, exception_predicate::message_contains("404 Not Found"))

so we assert that s3 server replies with the expected error.